bin/outputExtensionReplace.js

```diff
@@ -0,0 +1,69 @@
+const { access } = require('fs').promises;
+const { dirname, join, resolve } = require('path');
+
+async function checkFileExists(filePath) {
+  try {
+    await access(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+const shouldReplaceExtension = (value) =>
+  (value.startsWith('./') || value.startsWith('../')) &&
+  !value.endsWith('.mjs');
+
+/**
+ *
+ * @param {import('jscodeshift').FileInfo} file
+ * @param {import('jscodeshift').API} api
+ */
+module.exports = async function transformer(file, api) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  await Promise.all(
+    [
+      j.ImportDeclaration,
+      j.ExportNamedDeclaration,
+      j.ExportAllDeclaration,
+      j.ExportDefaultDeclaration,
+    ]
+      .flatMap((declaration) => root.find(declaration).nodes())
+      .flatMap(async (node) => {
+        const source = node.source;
+        if (!source) {
+          return;
+        }
+        const value = source.value;
+
+        if (shouldReplaceExtension(value)) {
+          const importPath = resolve(dirname(file.path), value);
+          const mjsPath = `${importPath}.mjs`;
+          const indexPath = join(importPath, 'index.mjs');
+
+          const mjsExists = await checkFileExists(mjsPath);
+          const indexExists = await checkFileExists(indexPath);
+
+          if (mjsExists) {
+            node.source.value = `${value}.mjs`;
+          } else if (indexExists) {
+            node.source.value = `${value}/index.mjs`;
+          } else {
+            console.error(
+              `missing ${mjsPath} and ${indexPath} for ${file.path}`,
+            );
+            return;
+          }
```

Comment:
ファイルの存在チェック、コスト高そうなイメージなので必要に応じてでも良いかも？
```suggestion
          const mjsExists = await checkFileExists(mjsPath);
          if (mjsExists) {
            node.source.value = `${value}.mjs`;
            return;
          }

          const indexExists = await checkFileExists(indexPath);
          if (indexExists) {
            node.source.value = `${value}/index.mjs`;
            return;
          }

          console.error(
              `missing ${mjsPath} and ${indexPath} for ${file.path}`,
          );
          return;
```

Reply:
@kc7891 
たしかにですね。以下のように変更加えることで、 `mjsPath` が存在したときは叩かなくなったかと思いますがいかがでしょうか👀

[refactor: checkFileExistsのタイミングを変更](https://github.com/openameba/spindle/pull/1178/commits/6e2adbd5d5065e092ecb046a07893c0200d62944)